internal/pkg/tfconf/module.go

```diff
@@ -122,20 +126,57 @@ func CreateModule(options *Options) (*Module, error) {
 		}
 	}
 
+	options.OutputValuesPath = os.Getenv("FOO")
+	// if OutputValuesPath is empty, set the default and generate the file
+	if options.OutputValues && options.OutputValuesPath == "" {
+		err := os.Setenv("FOO", "terraform-outputs.json")
+		if err != nil {
+			log.Fatal(err)
+		}
+		_, err = exec.Command("cd", options.Path, "&&", "terraform", "output", "--json", ">>", options.OutputValuesPath).Output()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	var terraformOutputs map[string]*TerraformOutput
+	if options.OutputValues {
+		byteValue, err := ioutil.ReadFile(options.OutputValuesPath)
+		if err != nil {
+			return nil, fmt.Errorf("caught error while reading the terraform outputs file at %s: %v", options.OutputValuesPath, err)
+		}
+		if err := json.Unmarshal(byteValue, &terraformOutputs); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Please extract these into a function, like `loadOutputValues`. The flow of code will be cleaner and it will be easier to test individual function and helps for mocking (if need be)

Reply:
All done~